src/main/java/edu/kit/provideq/toolbox/meta/setting/MetaSolverSettingDeserializer.java

```diff
@@ -0,0 +1,56 @@
+package edu.kit.provideq.toolbox.meta.setting;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MetaSolverSettingDeserializer extends JsonDeserializer<MetaSolverSetting> {
+    @Override
+    public MetaSolverSetting deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        ObjectCodec codec = jsonParser.getCodec();
+        JsonNode node = codec.readTree(jsonParser);
+
+        // Get type
+        MetaSolverSettingType type = MetaSolverSettingType.valueOf(node.get("type").asText());
+
+        // Create subclass based on the type
+        switch (type) {
+            case CHECKBOX -> {
+                return new Checkbox(
+                        node.get("name").asText(),
+                        node.get("state").asBoolean());
+            }
+            case RANGE -> {
+                return new Range(
+                        node.get("name").asText(),
+                        node.get("min").asDouble(),
+                        node.get("max").asDouble(),
+                        node.get("value").asDouble());
```

Comment:
I think Jackson should be able to deserialize polymorphic types when you tell it how to distinguish them. I can recommend the following guide (which I used in our PSE implementation): https://www.baeldung.com/jackson-inheritance

Reply:
Yeah I guess that works too. But do you think we should change the implementation at this point?